src/history/history.go

```diff
@@ -114,7 +117,13 @@ func SetState(repoURL string, cur State) error {
 	cur.Timestamp = time.Now()
 	var err error
 	if cur.Status != "deploying" || checkSubdomain(cur.Subdomain) {
-		history[repoURL].Current = &cur
+		if newTag := makeTag(repoURL, cur); newTag == tag {
+			history[repoURL].Current = &cur
+		} else {
+			err = errors.New("tag did not match")
```

Comment:
return `fmt.Errorf("old tag provided %s, does not match the original tag %s",oldtag,newtag)`